src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        return Storage.productStorage.stream()
+                .filter(product -> product.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        return Storage.productStorage;
+    }
+
+    @Override
+    public Product update(Product product) {
+        Product updatedProduct = get(product.getId()).get();
+        updatedProduct.setName(product.getName());
+        updatedProduct.setPrice(product.getPrice());
+        return updatedProduct;
+    }
```

Comment:
the logic of this method is to update all possible matches, not only the first one

let's create IntStream of indices, than filter out only those indices on which there are elements with the same id
then for each of the remaining indices, replace the element with our `product`

and let's return the new product (actually return the same object that we receive)